src/NetAppFiles/NetAppFiles/Snapshot/GetNetAppFilesSnapshot.cs

```diff
@@ -0,0 +1,133 @@
+{
+    [Cmdlet(
+        "Get",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesSnapshot",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(PSNetAppFilesSnapshot))]
+    [Alias("Get-AnfSnapshot")]
+    public class GetAzureRmNetAppFilesSnapshot : AzureNetAppFilesCmdletBase
+    {
+        [Parameter
+            (Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The resource group of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts",
+            nameof(ResourceGroupName))]
+        public string AccountName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the ANF pool", ParameterSetName = FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools",
+            nameof(ResourceGroupName),
+            nameof(AccountName))]
+        public string PoolName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools/volumes",
+            nameof(ResourceGroupName),
+            nameof(AccountName),
+            nameof(PoolName))]
+        public string VolumeName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The name of the ANF snapshot")]
+        [ValidateNotNullOrEmpty]
+        [Alias("SnapshotName")]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools/volumes",
+            nameof(ResourceGroupName),
+            nameof(AccountName),
+            nameof(PoolName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "The resource id of the ANF snapshot")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+        
+        [Parameter(
+            ParameterSetName = ParentObjectParameterSet,
+            Mandatory = false,
+            ValueFromPipeline = true,
+            HelpMessage = "The pool object containing the volume to return")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesPool VolumeInputObject { get; set; }
```

Comment:
@leonardbf nit: I would rename this to just `VolumeObject`

Reply:
Ok. Renamed